src/core/partitions.rs

```diff
@@ -67,217 +64,158 @@ impl From<&[u8]> for Partition {
     }
-    Ok(set)
+    Ok(res)
 }
 
-/// Generate all attribute values combinations from the given axes.
+/// Generates all cross-axes combinations of attribute values.
 ///
-/// - `current_axis`    : axis from which to start to combine values with other
-///   axes
-/// - `axes`            : list of axes
-/// - `map`             : map axes with their associated attribute values
+/// - `current_axis`            : axis for which to combine values with other axes
+/// - `axes`                    : list of axes
+/// - `attr_values_per_axis`    : map axes with their associated attribute values
 fn combine_attribute_values(
     current_axis: usize,
     axes: &[String],
-    map: &HashMap<String, Vec<u32>>,
-) -> Result<Vec<Vec<u32>>, Error> {
-    // get the current axis or return if there is no more axis
-    let axis = match axes.get(current_axis) {
-        None => return Ok(vec![]),
+    attr_values_per_axis: &HashMap<String, Vec<(u32, EncryptionHint)>>,
+) -> Result<Vec<(Vec<u32>, EncryptionHint)>, Error> {
+    let current_axis_name = match axes.get(current_axis) {
+        None => return Ok(vec![(vec![], EncryptionHint::Classic)]),
         Some(axis) => axis,
     };
-    // get the axes attribute value, wrapped into a `Vec`
-    let axis_values = map.get(axis).ok_or_else(|| {
-        Error::AttributeNotFound(format!(
-            "unexpected error: attribute values not found for axis: {}",
-            &axis
+
+    let current_axis_values = attr_values_per_axis.get(current_axis_name).ok_or_else(|| {
+        Error::CryptoError(format!(
+            "no attribute value found for axis: {current_axis_name}",
         ))
     })?;
 
-    // combine these values with all attribute values from the next axes
-    let other_values = combine_attribute_values(current_axis + 1, axes, map)?;
-    if other_values.is_empty() {
-        Ok(axis_values.iter().map(|v| vec![*v]).collect())
-    } else {
-        let mut combinations = Vec::with_capacity(axis_values.len() * other_values.len());
-        for av in axis_values {
-            for ov in &other_values {
-                let mut combined = Vec::with_capacity(1 + ov.len());
-                combined.push(*av);
-                combined.extend_from_slice(ov);
-                combinations.push(combined);
-            }
+    // Recursive call. Above checks ensure no empty list can be returned.
+    let other_values = combine_attribute_values(current_axis + 1, axes, attr_values_per_axis)?;
+
+    let mut combinations = Vec::with_capacity(current_axis_values.len() * other_values.len());
+    for (current_values, is_hybridized) in current_axis_values {
+        for (other_values, is_other_hybridized) in &other_values {
+            let mut combined = Vec::with_capacity(1 + other_values.len());
+            combined.push(*current_values);
+            combined.extend_from_slice(other_values);
+            combinations.push((combined, *is_hybridized | *is_other_hybridized));
```

Comment:
I didn't know the `|`  operator, isn't it clearer to use normal OR here `||`? It seems to be the same for `bool` but maybe I'm missing something (`||` seems more correct for me here)